x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java

```diff
@@ -726,7 +726,7 @@ public static Literal randomLiteral(DataType type) {
             case UNSIGNED_LONG, LONG, COUNTER_LONG -> randomLong();
             case DATE_PERIOD -> Period.of(randomIntBetween(-1000, 1000), randomIntBetween(-13, 13), randomIntBetween(-32, 32));
             case DATETIME -> randomMillisUpToYear9999();
-            case DATE_NANOS -> randomLong();
+            case DATE_NANOS -> randomLongBetween(0, Long.MAX_VALUE);
```

Comment:
This is unrelated, but has been wrong for a while.  I changed how we were logging dates in tests, which then triggered an issue with negative values here, thus needing to fix it now.

Reply:
Nit, could be simplified to `randomNonNegativeLong()`